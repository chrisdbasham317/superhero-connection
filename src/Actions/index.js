export const isLoading = bool => ({
  type: 'IS_LOADING',
  bool
});

export const hasErrored = msg => ({
  type: 'HAS_ERRORED',
  msg
});

export const setHeroes = heroes => ({
  type: 'SET_HEROES',
  heroes
})