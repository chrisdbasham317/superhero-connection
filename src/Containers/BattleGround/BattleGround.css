.section--battle-ground {
  background: url('https://image.freepik.com/free-vector/modern-comic-book-background-with-ray_1035-8002.jpg');
  background-position-y: center;
  background-size: cover;
  height: 94vh;
}


.form--hero-selection {
  align-items: center;
  border: 5px solid #ffffff;
  display: flex;
  flex-direction: column;
  height: 60vh;
  left: 35vw;
  padding: 2%;
  position: absolute;
  /* text-align: center; */
  top: 30vh;
  width: 35vw;
}

.h1--hero-form {
  color: #ffffff;
}

label {
  color: #ffffff;
  font-size: 1.4rem;
}

select {
  font-size: 1.5rem;
  height: 5vh;
  margin-bottom: 5vh;
  width: 70%;
}

.button--start-fight {
  border-radius: 20px;
  font-size: 1.25rem;
  height: 5vh;
  width: 50%;
}

.button--start-fight:hover {
  background: rgb(220, 145, 6);
  color: #ffffff;
  height: 6vh;
  width: 52%;
}

.section--heroes-selected {
  display: flex;
  justify-content: space-between;
}

.div--hero-1,
.div--hero-2 {
  max-width: 345px;
  width: 33vw;
}