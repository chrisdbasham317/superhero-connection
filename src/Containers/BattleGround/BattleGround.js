import React, { Component } from 'react';
import ReactModal from 'react-modal';
import { connect } from 'react-redux';
import { HeroCard } from '../../Components/HeroCard/HeroCard';
import './BattleGround.css';

export class BattleGround extends Component {
  constructor(combatant1, combatant2) {
    super(combatant1, combatant2);
    this.state = {
      challenger1: this.props.combatant1,
      challenger2: this.props.combatant2,
      modalState: true
    }
  }

  determineSelectStatus = (heroName, combatantNum) => {
    return this.props[`combatant${combatantNum}`] === heroName ? true : false;
  }
  render() {
    const hero1 = this.props.heroes.filter(hero => hero.id === this.props.combatant1);
    const hero2 = this.props.heroes.filter(hero => hero.id === this.props.combatant2);
    const renderReady = [...hero1, ...hero2];
    const options1 = this.props.heroes.map(hero => <option value={`${hero.name}`} selected={this.determineSelectStatus(hero.name, 1)}>{hero.name}</option>)
    const options2 = this.props.heroes.map(hero => <option value={`${hero.name}`} selected={this.determineSelectStatus(hero.name, 2)}>{hero.name}</option>)
    return (
      <section className='section--battle-ground'>
        <section className='section--heroes-modal'>
          <ReactModal
            ariaHideApp={false}
            isOpen={this.state.modalState}
            style={{
              overlay: {
                position: "fixed",
                top: 0,
                left: 0,
                right: 0,
                bottom: 0,
                backgroundColor: "rgba(0, 0, 0, 0.85)",
              },
              content: {}
            }}
            contentLabel="Hero Selections"
            className="HeroSelectionFormModal"
            overlayClassName="HeroSelectionFormOverlay"
          >
            <form className='form--hero-selection'>
              <h1>Select Your Heroes</h1>
              <select>{options1}</select>
              <select>{options2}</select>
              <button className='button-start-fight'>Fight!</button>
            </form>
          </ReactModal>
        </section>
        {renderReady.length > 1 ? <section className='section--heroes-selected'>
          <div>
            <HeroCard
              key={hero1[0].id}
              id={hero1[0].id}
              name={hero1[0].name}
              img={hero1[0].images.md}
            />
          </div>
          <div>
            <HeroCard
              key={hero2[0].id}
              id={hero2[0].id}
              name={hero2[0].name}
              img={hero2[0].images.md}
            />
          </div>
        </section> : null}
      </section>
    )
  }  
}

export const mapStateToProps = ({ heroes, combatant1, combatant2, toggleModal }) => ({
  heroes,
  combatant1,
  combatant2,
  toggleModal
})

export default connect(mapStateToProps)(BattleGround);