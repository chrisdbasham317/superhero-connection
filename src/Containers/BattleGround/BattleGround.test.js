import React from 'react';
import { shallow } from 'enzyme';
import { BattleGround, mapStateToProps, mapDispatchToProps } from './BattleGround';
import { setCombatant1, setCombatant2, setWinner } from '../../Actions';


jest.mock('../../Actions');

describe('BattleGround', () => {
  let wrapper
  const mockHeroes = [
    {
      id: 4,
      name: 'superman',
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 100,
      }
    },
    {
      id: 5,
      name: 'superboy',
      images: {
        md: 'medium image',
        lg: 'large image'
      },
      powerstats: {
        strength: 100,
        intelligence: 85,
      }
    }
  ]
  const mockSetCombatant1 = jest.fn();
  const mockSetCombatant2 = jest.fn();
  const mockSetWinner = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <BattleGround
        heroes={mockHeroes}
        combatant1={'superman'}
        combatant2={'superboy'}
        setCombatant1={mockSetCombatant1}
        setCombatant2={mockSetCombatant2}
        setWinner={mockSetWinner}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('should set the default battle to state if state is empty strings', () => {
    wrapper.setState({ challenger1: '', challenger2: '' });
    wrapper.instance().setDefaultBattle()
    expect(wrapper.state().challenger1).toEqual('Aquaman');
    expect(wrapper.state().challenger2).toEqual('Aquaman');
  });

  it('should call handleChange when a change occurs', () => {
    const mockHandleChange = jest.fn();
    wrapper.instance().handleChange = mockHandleChange
    const mockEvent = {
      target: {
        name: 'challenger1',
        value: 'Bane'
      }
    }
    wrapper.find('select').first().simulate('change', mockEvent);
    expect(mockHandleChange).toHaveBeenCalledWith(mockEvent);
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = {
      target: {
        name: 'challenger1',
        value: 'Bane'
      }
    };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state().challenger1).toEqual('Bane');
  });

  it('should call handleSubmit when submit btn is clicked', () => {
    const mockHandleSubmit = jest.fn();
    wrapper.instance().handleSubmit = mockHandleSubmit;
    const mockEvent = {
      preventDefault: jest.fn()
    };
    wrapper.find('.button--start-fight').simulate('click', mockEvent);
    expect(mockHandleSubmit).toHaveBeenCalledWith(mockEvent);
  });

  describe('handleSubmit', () => {
    let mockEvent

    beforeEach(() => {
      mockEvent = {
        preventDefault: jest.fn()
      }
    });
    it('should call setDefaultBattle when handleSubmit is called', () => {
      const mockSetDefaultBattle = jest.fn();
      wrapper.instance().setDefaultBattle = mockSetDefaultBattle;
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockSetDefaultBattle).toHaveBeenCalled();
    });

    it('should call setCombatant1 when handleSubmit is called', () => {
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockSetCombatant1).toHaveBeenCalledWith('superman');
    });

    it('should call setCombatant2 when handleSubmit is called', () => {
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockSetCombatant2).toHaveBeenCalledWith('superboy');
    }); 

    it('should update state when handleSubmit is called', async () => {
      await wrapper.instance().handleSubmit(mockEvent);
      expect(wrapper.state().modalState).toEqual(false);
    })

    it('should call determineWinner when handleSubmit is called', async () => {
      const mockDetermineWinner = jest.fn();
      wrapper.instance().determineWinner = mockDetermineWinner;
      await wrapper.instance().handleSubmit(mockEvent);
      expect(mockDetermineWinner).toHaveBeenCalled();
    });

    it('should call displayTimer when handleSubmit is called', async () => {
      const mockDisplayTimer = jest.fn();
      wrapper.instance().displayTimer = mockDisplayTimer;
      await wrapper.instance().handleSubmit(mockEvent);
      expect(mockDisplayTimer).toHaveBeenCalled();
    });
  });
})