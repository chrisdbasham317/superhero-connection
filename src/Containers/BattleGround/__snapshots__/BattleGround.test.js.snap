// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BattleGround should match the snapshot 1`] = `
<section
  className="section--battle-ground"
>
  <section
    className="section--heroes-modal"
  >
    <Modal
      ariaHideApp={false}
      bodyOpenClassName="ReactModal__Body--open"
      className="HeroSelectionFormModal"
      closeTimeoutMS={0}
      contentLabel="Hero Selections"
      isOpen={true}
      overlayClassName="HeroSelectionFormOverlay"
      parentSelector={[Function]}
      portalClassName="ReactModalPortal"
      role="dialog"
      shouldCloseOnEsc={true}
      shouldCloseOnOverlayClick={true}
      shouldFocusAfterRender={true}
      shouldReturnFocusAfterClose={true}
      style={
        Object {
          "content": Object {},
          "overlay": Object {
            "backgroundColor": "rgba(0, 0, 0, 0.85)",
            "bottom": 0,
            "left": 0,
            "position": "fixed",
            "right": 0,
            "top": 0,
          },
        }
      }
    >
      <form
        className="form--hero-selection"
      >
        <h1
          className="h1--hero-form"
        >
          Select Your Heroes
        </h1>
        <label
          htmlFor="challenger1"
        >
          Hero 1:
        </label>
        <select
          defaultValue="superman"
          name="challenger1"
          onChange={[Function]}
        >
          <option
            key="4"
            value="superman"
          >
            superman
          </option>
          <option
            key="5"
            value="superboy"
          >
            superboy
          </option>
        </select>
        <label
          htmlFor="challenger2"
        >
          Hero 2:
        </label>
        <select
          defaultValue="superboy"
          name="challenger2"
          onChange={[Function]}
        >
          <option
            key="4"
            value="superman"
          >
            superman
          </option>
          <option
            key="5"
            value="superboy"
          >
            superboy
          </option>
        </select>
        <button
          className="button--start-fight"
          onClick={[Function]}
        >
          Fight!
        </button>
      </form>
    </Modal>
  </section>
</section>
`;
